### PR TITLE
pdk: open_road_configuration: Get KLayout technology file and metal fill configuration from internal PDK setup

### PR DESCRIPTION
**Pull request by lpawelcz**
This PR modifies open_road_configuration so that it is now possible for a PDK dependency to define additional two fields:
* metal (density) fill configuration JSON file,
* KLayout technology LYT file.

The last one is required for GDS configuration so this field was marked as mandatory. The files are used by rules: `place_and_route` and `gds_write`. Default values specified for each PDK can be overwritten by attributes `density_config` and `klayout_lyt` of those rules.

It is reasonable to move handling of those files to the PDK dependencies setup because these are PDK-specific and can be reused for multiple designs.

CC @proppy

### CODE REVIEW

File: dependency_support/org_theopenroadproject_asap7/bundled.BUILD.bazel [hunk 1]
```diff
@@ -59,6 +59,7 @@ open_road_pdk_configuration(
         "M7": "0.5",
     },
     global_routing_signal_layers = "M2-M7",
+    klayout_tech_file = "@rules_hdl//dependency_support/org_theopenroadproject_asap7:asap7.lyt",
```
Comment (proppy) [inline dependency_support/org_theopenroadproject_asap7/bundled.BUILD.bazel:62]: q: no fill config by default in ASAP7?

**Review by proppy — APPROVED** [review]
/cc @QuantamHD

File: dependency_support/org_theopenroadproject_asap7/bundled.BUILD.bazel (hunk 1 above)
Comment (lpawelcz) [reply dependency_support/org_theopenroadproject_asap7/bundled.BUILD.bazel:62]: Looking at the flow examples in OpenROAD-flow-scripts - no, there is no `density_fill` stage in the flow targeting ASAP7 so there is no need for specifying the config file

**Comment by lpawelcz** [comment]
Hi @QuantamHD, please take a look at this PR. Merging this would unblock our work in https://github.com/google/xls/pull/1031

**Review by QuantamHD — APPROVED** [review]

**Comment by lpawelcz** [comment]
Thanks a lot!

**Pull request merged**